api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3624,23 +3625,67 @@ public void setObjectRetention(
       throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
           InternalException, InvalidBucketNameException, InvalidKeyException,
           InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
-    if (config == null) {
-      throw new IllegalArgumentException("null value is not allowed in config.");
+
+    this.setObjectRetention(
+        SetObjectRetentionArgs.builder()
+            .bucket(bucketName)
+            .object(objectName)
+            .versionId(versionId)
+            .config(config)
+            .overrideGovernance(bypassGovernanceRetention)
```

Comment:
As `bypassGovernanceMode()` is used in `removeObject()`, we would need to be consistent with names. Sorry for the confusion.